scripts/benchmark/checks/erc20_interact.py

```diff
@@ -0,0 +1,283 @@
+import logging
+import pathlib
+from typing import (
+    Tuple,
+)
+
+from abc import (
+    abstractmethod,
+)
+
+from web3 import (
+    Web3
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+    to_int,
+)
+
+from eth.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from eth.chains.base import (
+    MiningChain,
+)
+from eth.rlp.blocks import (
+    BaseBlock,
+)
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    SECOND_ADDRESS,
+    SECOND_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+)
+from utils.compile import (
+    get_compiled_contract
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+FIRST_TX_GAS_LIMIT = 1400000
+SECOND_TX_GAS_LIMIT = 60000
+TRANSFER_AMOUNT = 1000
+TRANSER_FROM_AMOUNT = 1
+
+CONTRACT_FILE = 'scripts/benchmark/contract_data/erc20.sol'
+CONTRACT_NAME = 'SimpleToken'
+
+W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
```

Comment:
We need to set *something* in order to prevent web3 trying to call out via RPC (which it isn't connect to anything).

```
Traceback (most recent call last):
  File "./scripts/benchmark/run.py", line 89, in <module>
    run()
  File "./scripts/benchmark/run.py", line 83, in run
    total_stat = total_stat.cumulate(benchmark.run(), increment_by_counter=True)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/base_benchmark.py", line 53, in run
    stat = self.execute()
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/erc20_interact.py", line 98, in execute
    lambda: self.mine_blocks(chain, self.num_blocks, self.num_tx)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/base_benchmark.py", line 58, in as_timed_result
    return time_call(fn)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/utils/meters.py", line 18, in time_call
    return_value = fn()
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/erc20_interact.py", line 98, in <lambda>
    lambda: self.mine_blocks(chain, self.num_blocks, self.num_tx)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/erc20_interact.py", line 116, in mine_blocks
    block = self.mine_block(chain, i, num_tx)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/erc20_interact.py", line 126, in mine_block
    self._apply_transaction(chain)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/erc20_interact.py", line 233, in _apply_transaction
    self._deploy_simple_token(chain)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/scripts/benchmark/checks/erc20_interact.py", line 136, in _deploy_simple_token
    w3_tx = SimpleToken.constructor().buildTransaction(W3_TX_DEFAULTS)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/utils/decorators.py", line 14, in _wrapper
    return self.method(obj, *args, **kwargs)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/contract.py", line 862, in buildTransaction
    return fill_transaction_defaults(self.web3, built_transaction)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/utils/transactions.py", line 55, in fill_transaction_defaults
    default_val = default_getter(web3, transaction)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/utils/transactions.py", line 26, in <lambda>
    'gas': lambda web3, tx: web3.eth.estimateGas(tx),
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/eth.py", line 303, in estimateGas
    [transaction],
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/manager.py", line 107, in request_blocking
    response = self._make_request(method, params)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/venv/lib/python3.6/site-packages/web3/manager.py", line 99, in _make_request
    params,

```

Reply:
For the stack trace you gave: one option is to move the `gas` value from https://github.com/ethereum/py-evm/pull/971/files#diff-373c9624e8e0573238de9774d078a2c6R143
up a line to https://github.com/ethereum/py-evm/pull/971/files#diff-373c9624e8e0573238de9774d078a2c6R136